os/hal/ports/KINETIS/LLD/USBHSv1/hal_usb_lld.c

```diff
@@ -427,6 +427,12 @@ void usb_lld_init(void) {
 #error USB clock setting not implemented for this KINETIS_MCG_MODE
 #endif /* KINETIS_MCG_MODE == ... */
 
+#if defined(K64F)
+  /* Switch from default MCGPLLCLK to IRC48M for USB */
+  //SIM->CLKDIV2 = SIM_CLKDIV2_USBDIV(0);
+  //SIM->SOPT2 |= SIM_SOPT2_PLLFLLSEL_IRC48M;

```

Comment:
So this is strange... I managed to get USB working by commenting out these lines, dropping the PLL clock frequency from 120MHz down to 96MHz, and setting `SIM_CLKDIV2_USBDIV` to divide it by 2. The weird part is that I found the following sentence in [AN4905](https://www.nxp.com/docs/en/application-note/AN4905.pdf#d51e5a1310_d11e168):

> The K64, K63, and K24 family devices **with 1 MByte of flash** were the first Kinetis MCUs to implement the crystal-less USB feature

(emphasis mine)

This is not mentioned anywhere else, and if it means what I think it means, then the MK64FX512 cannot use the IRC48M for crystal-less USB, but the MK64FX1M0 *can*.

Reply:
I'm going to have to leave this for someone smarter than me... as of now, USB works, but I have no idea if it's the best possible configuration.